pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1504,6 +1505,16 @@ func CheckNodeDiskPressurePredicate(pod *v1.Pod, meta algorithm.PredicateMetadat
 	return true, nil, nil
 }
 
+// CheckNodePIDPressurePredicate checks if a pod can be scheduled on a node
+// reporting pid pressure condition.
+func CheckNodePIDPressurePredicate(pod *v1.Pod, meta algorithm.PredicateMetadata, nodeInfo *schedulercache.NodeInfo) (bool, []algorithm.PredicateFailureReason, error) {
+	// check if node is under pid pressure
```

Comment:
Would it better to have a featuregating enabled here? Something like `utilfeature.DefaultFeatureGate.Enabled(features.TaintNodesByCondition)`

Reply:
Good point :) , we need to remove/disable this predicate if TaintNodesByCondition enabled.

xref https://github.com/kubernetes/kubernetes/pull/60398

Reply:
Done